### DISCUSSION TITLE
Static vs. dynamic library

### DISCUSSION THREAD

**Discussion by tomassliz** [pointfreeco/swift-composable-architecture#462] [Q&A] [answered]
Why is the TCA distributed as a static library? It was previously dynamic right?

**Comment by RabugenTom** [comment, marked as answer]
I think there's an explanation [here](https://github.com/pointfreeco/swift-composable-architecture/issues/70#issuecomment-627628374).

**Reply by tomassliz** [reply]
You're right. Thank you!

**Reply by stephencelis** [reply]
FWIW swift-composable-architecture doesn't declare itself as static or dynamic. We use the SPM default, which is static now, but Xcode 12.5 should automatically promote static libraries to be dynamic as necessary. You should definitely test this functionality in the beta, though, and report feedback as necessary.

**Reply by saroar** [reply]
here is other some reason which should we used (very good video for know about dynamic vs static )
https://www.youtube.com/watch?v=IqsKGyklmL0
